src/lib/OTA/OTA.cpp

```diff
@@ -183,4 +195,96 @@ void ICACHE_RAM_ATTR UnpackChannelDataSeqSwitches(volatile uint8_t* Buffer, CRSF
             break;
     }
 }
+#endif
 #endif // SEQ_SWITCHES
+
+#if (!defined HYBRID_SWITCHES_8 and !defined SEQ_SWITCHES) or defined UNIT_TEST
+
+#if TARGET_TX or defined UNIT_TEST
+
+void ICACHE_RAM_ATTR GenerateChannelData10bit(volatile uint8_t* Buffer, CRSF *crsf, uint8_t addr)
+{
+  uint8_t PacketHeaderAddr;
+  PacketHeaderAddr = (addr << 2) | RC_DATA_PACKET;
+  Buffer[0] = PacketHeaderAddr;
+  Buffer[1] = ((crsf->ChannelDataIn[0]) >> 3);
+  Buffer[2] = ((crsf->ChannelDataIn[1]) >> 3);
+  Buffer[3] = ((crsf->ChannelDataIn[2]) >> 3);
+  Buffer[4] = ((crsf->ChannelDataIn[3]) >> 3);
+  Buffer[5] = ((crsf->ChannelDataIn[0] & 0b110) << 5) |
+                           ((crsf->ChannelDataIn[1] & 0b110) << 3) |
+                           ((crsf->ChannelDataIn[2] & 0b110) << 1) |
+                           ((crsf->ChannelDataIn[3] & 0b110) >> 1);
+#ifdef One_Bit_Switches
+  Buffer[6] |= CRSF_to_BIT(crsf->ChannelDataIn[4]) << 7;
+  Buffer[6] |= CRSF_to_BIT(crsf->ChannelDataIn[5]) << 6;
+  Buffer[6] |= CRSF_to_BIT(crsf->ChannelDataIn[6]) << 5;
+  Buffer[6] |= CRSF_to_BIT(crsf->ChannelDataIn[7]) << 4;
+  Buffer[6] |= CRSF_to_BIT(crsf->ChannelDataIn[8]) << 3;
+  Buffer[6] |= CRSF_to_BIT(crsf->ChannelDataIn[9]) << 2;
+  Buffer[6] |= CRSF_to_BIT(crsf->ChannelDataIn[10]) << 1;
+  Buffer[6] |= CRSF_to_BIT(crsf->ChannelDataIn[11]) << 0;
+#endif
+}
+
+#elif TARGET_RX or defined UNIT_TEST
+
+void ICACHE_RAM_ATTR UnpackChannelData10bit(volatile uint8_t* Buffer, CRSF *crsf)
+{
+    crsf->PackedRCdataOut.ch0 = (Buffer[1] << 3) | ((Buffer[5] & 0b11000000) >> 5);
+    crsf->PackedRCdataOut.ch1 = (Buffer[2] << 3) | ((Buffer[5] & 0b00110000) >> 3);
+    crsf->PackedRCdataOut.ch2 = (Buffer[3] << 3) | ((Buffer[5] & 0b00001100) >> 1);
+    crsf->PackedRCdataOut.ch3 = (Buffer[4] << 3) | ((Buffer[5] & 0b00000011) << 1);
+#ifdef One_Bit_Switches
```

Comment:
There is a problem with the One_Bit_Switches def and it can be completely deleted since it is default for the normal RC packet.